File: UniversityApiBackend/DTOs/Students/StudentCreateDTO.cs
using System.ComponentModel.DataAnnotations;

namespace UniversityApiBackend.DTOs.Students
{
    public class StudentCreateDTO
    {
        public int Id { get; set; }
        [Required]
        public DateTime Dob { get; set; }

        public UserDTO User { get; set; }= new UserDTO();
        public List<StudentCourseDTO> Courses { get; set; }

        public AddressDTO Address { get; set; }= new AddressDTO();
    }
}

File: Frontend/front-university/src/app/services/api/models/studentCreateDTO.ts
import { AddressDto } from "./address-dto";
import { StudentCourseDTO } from "./student-dto";
import { User } from "./user";

export interface StudentCreateDTO {
  id?:      number;
  dob?:     Date;
  user?:    User;
  courses?: StudentCourseDTO[];
  address?: AddressDto;
}



